src/core/storage/sframe_interface/unity_sframe.cpp

```diff
@@ -101,8 +101,8 @@ void unity_sframe::construct_from_sframe_index(std::string location) {
 
   auto status = fileio::get_file_status(location);
 
-  if ((status.first == fileio::file_status::FS_UNAVAILABLE ||
-       status.first == fileio::file_status::MISSING) &&
+  // web protocol is a bad name, it should mean read-only protocol
+  if (status.first == fileio::file_status::REGULAR_FILE &&
```

Comment:
Shouldn't this test for both regular file and missing?  Why do we ignore the missing path?

Reply:
I actually don't get why it works in the repro case when only testing for regular. The repro involves a 404 at the directory path -- I would think this would come in as MISSING.

Reply:
for remote fs, unlike hdfs or s3, http/https protocol is not handled eagerl, and the URI will be marked as regular file instead of marking it as MISSING or UNAVAILABLE (at least this is what original `get_file_status` does). 

Now we can understand, regular file can either be eagerly verified (s3, local file), or can be file behind protocol like http/https, whose real file status verification is deferred until reading.

Reply:
That's really strange. Was this always the behavior? How did the regression happen on this issue, and what code change caused it?

Reply:
Yeah. The regression is that I modified this file by moving http probing logic into this deleted section. 

The original behavior is `try_open_file` blindly regardless the file status returned from `get_file_status`, 
```
auto status = fileio::get_file_status(location);
  if (fileio::is_web_protocol(fileio::get_protocol(location))) {
    // if it is a web protocol, we cannot be certain what type of file it is.
    // HEURISTIC:
    //   assume it is a "directory" and try to load dir_archive.ini
    //   if we can open it, it is a regular file. Otherwise not.
    if (fileio::try_to_open_file(location + "/dir_archive.ini")) {
      status = fileio::file_status::DIRECTORY;
    } else {
      status = fileio::file_status::REGULAR_FILE;
    }
  }
```
it discards the info of MISSING and FS_UNAVAILABLE, which is problematic.

In this PR, 
1. for MISSING and FS_UNAVAILABLE, we can skip probing which saves network. 
2. For DIRECTORY, no need to probe. 
3. For REGULAR file, if it's so called `web_protocol`, we need to probe its real file status eagerly through network. For s3 or hdfs, the really file status is already probed through network.

Essentially, we should move the probing into `get_file_status`. But here, let's get it in.

Reply:
@hoytak No need to handle MISSING or UNAVAILABLE since these file statuses are verified eagerly through network already.